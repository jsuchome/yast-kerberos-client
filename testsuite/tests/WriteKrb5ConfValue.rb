# encoding: utf-8

#  WriteKrb5ConfValue.ycp
#  Test of Kerberos::WriteKrb5ConfValue function
#  Author:	Jiri Suchomel <jsuchome@suse.cz>
#  $Id$
module Yast
  class WriteKrb5ConfValueClient < Client
    def main
      # testedfiles: Kerberos.ycp
      Yast.import "Testsuite"
      Yast.import "Kerberos"

      @p = path(".etc.krb5_conf.v.libdefaults.default_realm")

      Testsuite.Test(lambda { Kerberos.WriteKrb5ConfValue(@p, "SUSE") }, [
        {},
        {},
        {}
      ], 0)

      Testsuite.Test(lambda { Kerberos.WriteKrb5ConfValue(@p, "") }, [{}, {}, {}], 0)

      Testsuite.Test(lambda { Kerberos.WriteKrb5ConfValue(@p, nil) }, [
        {},
        {},
        {}
      ], 0)

      @p = path(".etc.krb5_conf.v.SUSE.kdc")
      Testsuite.Test(lambda do
        Kerberos.WriteKrb5ConfValues(@p, ["kdc.suse.cz", "kdc.suse.de"])
      end, [
        {},
        {},
        {}
      ], 0)

      Testsuite.Test(lambda do
        Kerberos.WriteKrb5ConfValuesAsString(@p, "kdc.suse.cz kdc.suse.de")
      end, [
        {},
        {},
        {}
      ], 0)

      nil
    end
  end
end

Yast::WriteKrb5ConfValueClient.new.main
