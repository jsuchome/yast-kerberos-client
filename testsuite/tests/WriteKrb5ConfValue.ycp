/**
 *  WriteKrb5ConfValue.ycp
 *  Test of Kerberos::WriteKrb5ConfValue function
 *  Author:	Jiri Suchomel <jsuchome@suse.cz>
 *  $Id$
 */

{
    // testedfiles: Kerberos.ycp
    import "Testsuite";
    import "Kerberos";

    path p = .etc.krb5_conf.v.libdefaults.default_realm;

    Testsuite::Test(``(Kerberos::WriteKrb5ConfValue (p, "SUSE")), [ $[], $[], $[] ], 0);

    Testsuite::Test(``(Kerberos::WriteKrb5ConfValue (p, "")), [ $[], $[], $[] ], 0);

    Testsuite::Test(``(Kerberos::WriteKrb5ConfValue (p, nil)), [ $[], $[], $[] ], 0);

    p = .etc.krb5_conf.v.SUSE.kdc;
    Testsuite::Test(``(Kerberos::WriteKrb5ConfValues (p, ["kdc.suse.cz", "kdc.suse.de"])), [ $[], $[], $[] ], 0);

    Testsuite::Test(``(Kerberos::WriteKrb5ConfValuesAsString (p, "kdc.suse.cz kdc.suse.de")), [ $[], $[], $[] ], 0);
}
