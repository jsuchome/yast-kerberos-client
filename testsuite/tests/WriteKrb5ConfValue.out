Write	.etc.krb5_conf.v.libdefaults.default_realm ["SUSE"] true
Return	true
Return	false
Return	false
Write	.etc.krb5_conf.v.SUSE.kdc ["kdc.suse.cz", "kdc.suse.de"] true
Return	true
Write	.etc.krb5_conf.v.SUSE.kdc ["kdc.suse.cz", "kdc.suse.de"] true
Return	true
